### PR TITLE
Add an .editorconfig and do some cleanup of leading whitespace

### PR DESCRIPTION
**Pull request by aeischeid**
I noticed some of the artifacts created by the quickstart script had an odd mix of spaces and tabs for indents. 

https://editorconfig.org/ is meant to help out with this sort of thing, so I added one of those based on what it appears most of the files contained already. This should help things become more consistent over time.

Looked like the predominant indent choice was tabs so I set the config to be that, and updated a handful of the inconsistencies

Though the diff looks like kind of a lot it is all just indent white space, and I was careful to avoid breaking spots where spaces were used to vertically align.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+[*]
+indent_style = tab
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{md}]
+trim_trailing_whitespace = false
```
Comment (aeischeid) [inline .editorconfig]: might want to avoid this for `.adoc` files too?

**Comment by ddelponte** [comment]
Thanks again @aeischeid for this contribution!

Would you be so kind as to rebase against master?  There have been quite a few changes to master since this PR was created.

Rebasing should fix the failing checks.

**Comment by aeischeid** [comment]
rebased, but looks like one test failing still. hm. not sure how the whitespace would have caused it...

**Comment by sdelamo** [comment]
Closed via: https://github.com/grails-plugins/grails-spring-security-core/pull/609

**Pull request closed without merging**